src/Kmd.Logic.FileSecurity.Client/FileSecurityClient.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Kmd.Logic.FileSecurity.Client.ServiceMessages;
+using Kmd.Logic.Identity.Authorization;
+using Microsoft.Rest;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Kmd.Logic.FileSecurity.Client
+{
+    /// <summary>
+    /// Class to use the autogenerated client class to call APIs
+    /// </summary>
+    public sealed class FileSecurityClient : LogicHttpClientProvider
+    {
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FileSecurityClient"/> class.
+        /// </summary>
+        /// <param name="httpClient">The HTTP client to use. The caller is expected to manage this resource and it will not be disposed.</param>
+        /// <param name="tokenProviderFactory">The Logic access token provider factory.</param>
+        /// <param name="options">The required configuration options.</param>
+        public FileSecurityClient(HttpClient httpClient, LogicTokenProviderFactory tokenProviderFactory, FileSecurityOptions options)
```

Comment:
I would use `ITokenProviderFactory` abstraction. See this PR https://github.com/kmdlogic/kmd-logic-cpr-client/commit/42107d6d5f72671be6f2ab26dfa80fef45677475#diff-1a390da90b6b76ed9245ecc74bcc270b451723b95d3a031ec03d6d4528c94f1f

Reply:
Used `ITokenProviderFactory` as suggested.